src/Sulu/Bundle/AdminBundle/Resources/js/components/CircleSelection/CircleSelectionRenderer.js

```diff
@@ -0,0 +1,196 @@
+// @flow
+import React from 'react';
+import type {Normalizer, SelectionData} from './types';
+import ModifiableCircle from './ModifiableCircle';
+import PositionNormalizer from './normalizers/PositionNormalizer';
+import RoundingNormalizer from './normalizers/RoundingNormalizer';
+import SizeNormalizer from './normalizers/SizeNormalizer';
+
+type Props = {
+    containerHeight: number,
+    containerWidth: number,
+    disabled: boolean,
+    filled: boolean,
+    label?: string,
+    maxRadius?: number,
+    minRadius?: number,
+    onChange: (value: ?SelectionData) => void,
+    percentageValues: boolean,
+    resizable: boolean,
+    round: boolean,
+    value: SelectionData | typeof undefined,
+};
+
+export default class CircleSelectionRenderer extends React.Component<Props> {
+    static defaultProps = {
+        disabled: false,
+        filled: false,
+        percentageValues: false,
+        resizable: true,
+        round: true,
+    };
+
+    handleChange = (value: ?SelectionData) => {
+        const {onChange, containerWidth, containerHeight, percentageValues} = this.props;
+
+        if (!percentageValues || !value) {
```

Comment:
I think a `hoc` would fit best here, since you can use a RectangleSelection with percentageValues also outside of an ImageMap. But on the other hand it's probably very complicated to create a reusable, flexible `hoc` for that. WDYT?

Reply:
Why do you think it would be so complicated? Which `props` would it have to be aware of? I can think of the following:

- containerWidth
- containerHeight
- onChange
- value

And these would be more or less the same for both kind of selections, wouldn't they?

Reply:
But the `value` and the value of the first argument of `onChange` differ a lot between a circle and a rectangle e.g. ... Also, that hoc would need to know, that `value.top` should be divided by `containerHeight` and `value.radius` by `containerWidth` e.g. ... should that all be hardcoded?

Reply:
Yeah, I would say that this HOC can be tied to the `withContainerSize` HOC, and therefore can expect that there is a `containerHeight` and `containerWidth` attribute. But you are right, the `value` itself is different... I see two solutions to this:

1. We introduce a HOC that takes additional arguments that somehow tell which arguments have to be transformed.
2. We introduce two HOCs that are specific for the `RectangleSelection` and `CircleSelection`. Both of these HOCs could still take care of the transforming the `onChange` and `value` props, and adapt to the characteristics of the component they are wrapping. Then these component would have to be in the `RectangleSelection` and `CircleSelection` folder. While there would be less code reuse, I think the code would still be easier to understand, because the transforming from percentages to pixels and vice versa would take place in less places.

Reply:
I would prefer option 2

Reply:
Sounds good to me, go for it :slightly_smiling_face: